packages/server/lib/socket.js

```diff
@@ -71,7 +73,17 @@ class Socket {
     })
   }
 
-  watchTestFileByPath (config, specConfig, options) {
+  onStudioTestFileChange (filePath) {
+    return this.onTestFileChange(filePath).then(() => {
```

Comment:
may be worth adding a comment? was non-obvious to me what this was doing
```suggestion
    // wait for the studio test file to be written to disk, then reload the test
    return this.onTestFileChange(filePath).then(() => {
```

Reply:
I've added some additional comments to clarify things